packages/web3-providers-ipc/src/index.ts

```diff
@@ -84,17 +97,15 @@ export default class IpcProvider<API extends Web3APISpec = EthExecutionAPI> exte
 	}
 
 	protected _onCloseEvent(event: CloseEvent): void {
-		if (
-			this._reconnectOptions.autoReconnect &&
-			(![1000, 1001].includes(event.code) || !event.wasClean)
-		) {
+		if (!event && this._reconnectOptions.autoReconnect) {
```

Comment:
There can be normal close events are those considered as well ? 
so this condition should only be true if there is no clean close or event codes are not 1000 and 1001,

Reply:
IPC providers use `import { Socket } from 'net';` to connect. 
in the close event of Socket we do not have any `code` or `wasClean` information.
That is why for IPC we have a different condition.

but when we call the `disconnect` method we create an event on our own. and in this case, we will have an event object here.

Reply:
ok, so is this tested for normal close, its existing without reconnect?

Reply:
yes, this is a test "exit without reconnection":
```ts 
                 it('should emit connect and disconnected events', async () => {
			const server = await startGethServer(8547);
			const web3Provider = new IpcProvider(server.path);
			expect(!!(await waitForEvent(web3Provider, 'connect'))).toBe(true);
			const disconnectPromise = waitForEvent(web3Provider, 'disconnect');
			web3Provider.disconnect();
			expect(!!(await disconnectPromise)).toBe(true);
			await server.close();
		});
```

Reply:
ok